src/Parquet.chpl

```diff
@@ -11,8 +13,34 @@ module Parquet {
   extern var ARROWINT64: c_int;
   extern var ARROWINT32: c_int;
   extern var ARROWUNDEFINED: c_int;
+  extern var ARROWERROR: c_int;
 
   enum ArrowTypes { int64, int32, notimplemented };
+
+  record parquetErrorMsg {
+    var errMsg: c_ptr(uint(8));
+    proc init() {
+      errMsg = c_nil;
+    }
+    
+    proc deinit() {
+      extern proc c_free_string(ptr);
+      c_free_string(errMsg);
+    }
+
+    proc parquetError(lineNumber, routineName, moduleName) throws {
+      extern proc strlen(a): int;
+      var err;
+      try! err = createStringWithNewBuffer(errMsg, strlen(errMsg));
+      // free error message before throw since it can't be used again
```

Comment:
I don't see the free explicitly called here, is that ok?  Presumably `deinit` gets invoked on exit?  Otherwise maybe you could use a `defer`?

Reply:
Ya, `deinit` is called on exit, so that should handle the freeing, I will remove that comment, that is a residual from an intermediate step that came in this development; thanks for pointing that out.